worker/lease/secretaries.go

```diff
@@ -0,0 +1,94 @@
+package lease
+
+import (
+	"time"
+
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/core/lease"
+)
+
+// SingularSecretary implements Secretary to restrict claims to either
+// a lease for the controller or a specific model, holdable only by machine-tag
+// strings.
+//
+// It would be nicer to have a single controller-level component managing all
+// singular leases for all models -- and thus be able to validate that
+// proposed holders really are env managers -- but the complexity of threading
```

Comment:
env managers? that's so juju 1.x

Reply:
Yeah, this code was mostly just move wholesale from state. I'll update the comment.